ADEN-4833 Add src=rec for PageFair ads

dianafa opened the pull request:
https://wikia-inc.atlassian.net/browse/ADEN-4833

dianafa committed:
add src=rec param to slots recovered by PF or SP

dianafa committed:
ADEN-4833

fix tests

dianafa committed:
add rec param only when detection is active

Commit:
ADEN-4833 Fix and refactor existing test

Commit:
ADEN-4833 Add tests

nandy-andy commented:
The code changes are readable and I can read from them that it's going to do what we want it to do. It's covered with unit tests :+1: ...but I wonder why we did it differently than SourcePoint. I'd like to take a 2nd look tomorrow. Maybe we'll pair-CR, @dianafa? :-) Oh, and Jenkins failed :-(

dianafa committed:
Merge branch 'dev' into ADEN-4833

dianafa commented:
Jenkins fails not related to this PR.

Generally the reason was that I didn't like the way we structured SourcePoint recovery code. Before it was a general "recovery", not prepared to handle more than one recovery provider.
I didn't want to extend `recoverySlots` array or create a new one, for PF recovery in provider/directGpt.js (https://github.com/Wikia/app/pull/12567/files#diff-17d2543e01f9dbebdd79cc24d8a5674eR55) that is general gpt file, with configuration specific for pageFair. I'm not sure if it's clear what I mean though, let's pair-cr on that :)

dianafa committed:
ADEN-4833

make SP consistent with PF: keep recoverableSlots in ext.wikia.aRecoveryEngine.sourcePoint, change recoverableSlots to array, add tests

dianafa committed:
ADEN-4833

remove unused ext.wikia.aRecoveryEngine.sourcePoint imports and change gptCanBeLoaded in googleTag

dianafa committed:
ADEN-4833 change filename and module name

dianafa committed:
ADEN-4833

add isSourcePointRecoverable to pushAd

nandy-andy commented on extensions/wikia/ARecoveryEngine/js/PageFair/pageFair.js:
```diff
@@ -19,19 +20,27 @@ define('ext.wikia.aRecoveryEngine.recovery.pageFair', [
+	function isBlocking() {
+		var isBlocking = !!(win.ads && win.ads.runtime.pf && win.ads.runtime.pf.blocking);
+
+		log(['isBlocking', isBlocking], 'debug', logGroup);
```
We can also replace `'debug'` in all places with `log.levels.debug` as :cherries: ;-)

dianafa replied:
thanks! :)

nandy-andy commented on extensions/wikia/ARecoveryEngine/js/PageFair/pageFair.js:
```diff
@@ -19,19 +20,27 @@ define('ext.wikia.aRecoveryEngine.recovery.pageFair', [
 
 	function isSlotRecoverable(slotName) {
-		var result = isPageFairRecoveryEnabled() && !!recoverableSlots[slotName];
+		var result = isPageFairRecoveryEnabled() && recoverableSlots.indexOf(slotName) !== -1;
```
What do you think of renaming `isPageFairRecoveryEnabled()` to just `isEnabled()`?

nandy-andy commented on extensions/wikia/ARecoveryEngine/js/sourcePoint.js:
```diff
@@ -51,12 +60,17 @@ define('ext.wikia.aRecoveryEngine.recovery.sourcePointHelper', [
-	function isSourcePointRecoverable(slotName, recoverableSlots) {
-		return isSourcePointRecoveryEnabled() && recoverableSlots.indexOf(slotName) !== -1;
+	function isSlotRecoverable(slotName) {
+		var result = isSourcePointRecoveryEnabled() && recoverableSlots.indexOf(slotName) !== -1;
```
What do you think of renaming `isSourcePointRecoveryEnabled()` to just `isEnabled()`? :-p

nandy-andy commented on extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getBootstrap.mustache:
```diff
@@ -19,7 +19,7 @@
 					document.getElementById('WikiaArticleMsg').style.display = 'block';
 					window.addEventListener('load', function () {
-						require(['ext.wikia.aRecoveryEngine.recovery.sourcePointHelper'], function (helper) {
+						require(['ext.wikia.aRecoveryEngine.recovery.sourcePoint'], function (helper) {
```
We'll most likely need a fallback until cache re-validates.

nandy-andy commented on extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getBootstrap.mustache:
```diff
@@ -37,7 +37,7 @@
 
 			window.addEventListener('load', function () {
-				require(['ext.wikia.aRecoveryEngine.recovery.sourcePointHelper'], function (helper) {
+				require(['ext.wikia.aRecoveryEngine.recovery.sourcePoint'], function (helper) {
```
What will happen if a page will be cached and ask for module which doesn't exist anymore?

nandy-andy commented on extensions/wikia/AdEngine/js/lookup/lookupFactory.js:
```diff
@@ -52,7 +52,7 @@ define('ext.wikia.adEngine.lookup.lookupFactory', [
 			called = true;
 
-			helper.addOnBlockingCallback(onResponseCallbacks.start);
+			recoveryHelper.addOnBlockingCallback(onResponseCallbacks.start);
```
How does PF cooperates with RTBs?

nandy-andy commented on extensions/wikia/AdEngine/js/SlotTweaker.js:
```diff
@@ -3,11 +3,10 @@ define('ext.wikia.adEngine.slotTweaker', [
 	'wikia.log',
 	'wikia.window'
-], function (DOMElementTweaker, messageListener, adSlot, recoveryHelper, doc, log, win) {
+], function (DOMElementTweaker, messageListener, adSlot, doc, log, win) {
```
Unused dependency, good catch :+1:

nandy-andy commented on extensions/wikia/AdEngine/js/context/adContext.js:
```diff
@@ -111,9 +111,10 @@ define('ext.wikia.adEngine.adContext', [
+
 			context.opts.sourcePointDetectionMobile = (context.targeting.skin === 'mercury' &&
-			geo.isProperGeo(instantGlobals.wgAdDriverSourcePointDetectionMobileCountries));
+				geo.isProperGeo(instantGlobals.wgAdDriverSourcePointDetectionMobileCountries));
```
Is it an unnecessary tab?

dianafa replied:
imo is necessary

nandy-andy commented on extensions/wikia/AdEngine/js/lookup/lookupFactory.js:
```diff
@@ -2,10 +2,10 @@
 	'wikia.lazyqueue',
 	'wikia.log'
-], function (adContext, adTracker, helper, lazyQueue, log) {
+], function (adContext, adTracker, recoveryHelper, lazyQueue, log) {
```
What do you think of changing `recoveryHelper` in the functions arguments list to `sourcePoint` or `sourcePointRecovery`?

dianafa replied:
yep, was thinking about that

nandy-andy commented on extensions/wikia/AdEngine/js/provider/gpt/googleTag.js:
```diff
@@ -62,10 +62,11 @@ define('ext.wikia.adEngine.provider.gpt.googleTag', [
 			node = doc.getElementsByTagName('script')[0],
-			// load GPT when API not ready and recovery is non blocking or recovery is pageFair (never blocking)
-			pageFairRecovery = pageFair && pageFair.isPageFairRecoveryEnabled(),
+			pageFairRecoveryisBlocking = pageFair && pageFair.isBlocking(),
```
How about `pageFairRecoveryisBlocking` -> `pageFairRecoveryIsBlocking` or `isPageFairRecoveryBlocking`?

dianafa replied:
better: `isPageFairBlocking` - it's not a recovery that blocks. PF just detects it. Generally I don't like any of those names...

dianafa committed:
ADEN-4833

change isPageFairRecoveryEnabled and isSourcePointRecoveryEnabled to isEnabled

dianafa committed:
replace debug with log.levels.debug

dianafa committed:
ADEN-4833

rename recoveryHelper to sourcePoint

dianafa committed:
ADEN-4833

unify pageFair & sourcePoint recovery on off

dianafa committed:
ADEN-4833

remove old comment, change var name

dianafa committed:
ADEN-4833

add fallback dependency

dianafa committed:
ADEN-4833

add wgAdDriverPageFairRecoveryCountries to instantGlobals

dianafa committed:
Merge branch 'dev' into ADEN-4833

dianafa committed:
ADEN-4833

require floatingMedrec as optional & remove unneded var

mleczey commented on extensions/wikia/ARecoveryEngine/js/sourcePoint.js:
```diff
@@ -43,20 +52,25 @@ define('ext.wikia.aRecoveryEngine.recovery.sourcePointHelper', [
 	function isBlocking() {
-		log(['isBlocking', !!(win.ads && win.ads.runtime.sp && win.ads.runtime.sp.blocking)], 'debug', logGroup);
-		return isSourcePointRecoveryEnabled() && !!(win.ads && win.ads.runtime.sp && win.ads.runtime.sp.blocking);
+		var isBlocking = !!(win.ads && win.ads.runtime.sp && win.ads.runtime.sp.blocking);
```
Just checking, I assume we don't need `isEnabled` anymore here?

dianafa replied:
yes, we sometimes want to check if user is blocking even though recovery is enabled

mleczey commented on extensions/wikia/AdEngine/js/lookup/prebid/prebid.js:
```diff
@@ -21,13 +21,6 @@ define('ext.wikia.adEngine.lookup.prebid', [
-	 * remember about the additional [320, 480] slot size, see:
-	 * https://github.com/Wikia/app/pull/12269/files#diff-5bbaaa809332f9adaddae42c8847ae5bR6015
-	 */
-
```
I think this change is not related. Can we get rid of it?

dianafa replied:
I just encountered this old piece of code and I want to clean this up :)

mleczey commented on extensions/wikia/AdEngine/js/provider/gpt/helper.js:
```diff
@@ -91,7 +99,7 @@ define('ext.wikia.adEngine.provider.gpt.helper', [
 			}
 
-			if (shouldPushRecoverableAd) {
+			if (isRecoveryEnabled && isBlocking && adIsRecoverable) {
```
Would it be possible to extract method here, or simplify conditions logic in changed lines in this file?

dianafa committed:
ADEN-4833

simplify logic

mleczey commented:
Sorry, I just copied wrong link to pull request. Reference added by me has nothing to do with this.

dianafa committed:
ADEN-4833

load gpt when recovery is pagefair & use log.levels.XX
